### PR TITLE
Tx2 install dev

### PR DESCRIPTION
**Pull request by Pavlos-Tosidis**
This PR introduces the instructions for running OpenDR toolkit on an embedded device.
The docker images are uploaded on dockerhub and are thoroughly tested. 
A Dockerfile-embedded is also included for future reference/baseline to build the docker image as well as the install_nvidia.sh which installs all dependencies correctly.
#195 should also be closed since this PR is the improved version.

### CODE REVIEW

File: Dockerfile-embedded [hunk 1]
```diff
@@ -0,0 +1,201 @@
+#
+# this dockerfile roughly follows the 'Installing from source' from:
```
Comment (passalis) [inline Dockerfile-embedded:2]:
```suggestion
# This dockerfile roughly follows the 'Installing from source' ROS instructions from:
```

File: Dockerfile-embedded [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            qv4l2 \
+            v4l-utils \
+            zlib1g-dev
+
+RUN apt-get install ca-certificates -y
+
+# OpenCV looks for the cuDNN version in cudnn_version.h, but it's been renamed to cudnn_version_v8.h
+RUN ln -s /usr/include/$(uname -i)-linux-gnu/cudnn_version_v8.h /usr/include/$(uname -i)-linux-gnu/cudnn_version.h
+
+# architecture-specific build options
```
Comment (passalis) [inline Dockerfile-embedded:77]:
```suggestion
# Architecture-specific build options
```

File: Dockerfile-embedded [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+ARG CUDA_ARCH_BIN=""
+ARG ENABLE_NEON="OFF"
+
+# clone and configure OpenCV repo
```
Comment (passalis) [inline Dockerfile-embedded:81]:
```suggestion
# Clone and configure OpenCV repo
```

File: Dockerfile-embedded [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+RUN cd opencv/build && make install
+RUN cd opencv/build && make package
+
+RUN cd opencv/build && tar -czvf OpenCV-${OPENCV_VERSION}-$(uname -i).tar.gz *.deb
+
+RUN apt-get update -y
+RUN apt-get install software-properties-common -y && \
+    apt-get update
+
+# add the ROS deb repo to the apt sources list
```
Comment (passalis) [inline Dockerfile-embedded:131]:
```suggestion
# Add the ROS deb repo to the apt sources list
```

File: Dockerfile-embedded [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+		gnupg2 \
+		lsb-release \
+		ca-certificates \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+RUN curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | apt-key add -
+
+#
+# install bootstrap dependencies
```
Comment (passalis) [inline Dockerfile-embedded:149]:
```suggestion
# Install bootstrap dependencies
```

File: Dockerfile-embedded [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+          python3-rosinstall-generator \
+          python3-vcstool \
+          libyaml-cpp-dev \
+          build-essential && \
+    rosdep init && \
+    rosdep update && \
+    rm -rf /var/lib/apt/lists/*
+
+#
+# download/build the ROS source
```
Comment (passalis) [inline Dockerfile-embedded:164]:
```suggestion
# Download/build the ROS source
```

File: docs/reference/installation.md [hunk 7]
```diff
@@ -1,9 +1,9 @@
 # Installing OpenDR toolkit
 
 OpenDR can be installed in the following ways:
-1. Using *pip* (CPU/GPU support)
-2. Using *docker* (CPU/GPU support)
-3. By cloning this repository (CPU/GPU support, for advanced users only)
+1. By cloning this repository (CPU/GPU support)
```
Comment (passalis) [inline docs/reference/installation.md:4]: Please revert this change.

File: docs/reference/installation.md [hunk 8]
```diff
@@ -196,3 +196,100 @@ If you plan to use GPU-enabled functionalities, then you are advised to install
 **HINT:** All tests probe for the `TEST_DEVICE` enviromental variable when running.
 If this enviromental variable is set during testing, it allows for easily running all tests on a different device (e.g., setting `TEST_DEVICE=cuda:0` runs all tests on the first GPU of the system).
 
+## Build the docker images yourself _(optional)_
```
Comment (passalis) [inline docs/reference/installation.md:199]: Please remove this section and appropriately update `customize.md` file instead by adding only the instructions for NVIDIA embedded devices.

**Review by passalis — CHANGES_REQUESTED** [review]
Thank you! I have added a few comments regarding the documentation. `develop` has a separate documentation file where instructions for customization/building docker images is described, so the relevant instructions should be added there instead.

**Comment by passalis** [comment]
Please also update `CHANGELOG.md` to indicate that we have included instructions, an improved embedded dockerfile and changes to support these devices.

File: docs/reference/installation.md (hunk 8 above)
Comment (Pavlos-Tosidis) [reply docs/reference/installation.md]: Added the section to the appropriate file.

**Review by ad-daniel — APPROVED** [review]
Thank you!

**Review by passalis — APPROVED** [review]
Thank you!
